Author: aszlig <aszlig@nix.build>
Date:   Tue Mar 27 05:37:08 2018 +0200

    Fix parsing port number in URLs during redirection
    
    The parsing in the redirects module was simply determining the base host
    via trimming off everything up to //, so something like
    http://localhost:3000 will result in localhost:3000, which isn't a valid
    host name.
    
    Signed-off-by: aszlig <aszlig@nix.build>
    Filename: redirect-fix-port.patch

diff --git a/website/server/middlewares/redirects.js b/website/server/middlewares/redirects.js
index 9bc40cb43..0de427c2e 100644
--- a/website/server/middlewares/redirects.js
+++ b/website/server/middlewares/redirects.js
@@ -1,11 +1,11 @@
 import nconf from 'nconf';
+import url from 'url';
 
 const IS_PROD = nconf.get('IS_PROD');
 const IGNORE_REDIRECT = nconf.get('IGNORE_REDIRECT') === 'true';
 const BASE_URL = nconf.get('BASE_URL');
 
-let baseUrlSplit = BASE_URL.split('//');
-const BASE_URL_HOST = baseUrlSplit[1];
+const BASE_URL_HOST = url.parse(BASE_URL).hostname;
 
 function isHTTP (req) {
   return ( // eslint-disable-line no-extra-parens
